### PR TITLE
Update gnupg setup docs

### PR DESCRIPTION
**Pull request by benjamineskola**
People having trouble getting GPG working for access to `govuk-secrets` seems to be a recurring issue, so I'm trying to fill in some of the gaps in the documentation.

https://trello.com/c/VZyB4RIu/1242-gpg-secrets-documentation-out-of-date

### CODE REVIEW

**Comment by benjamineskola** [comment]
An additional potential change is to replace the recommendation of `keyserver.ubuntu.com` with `keys.openpgp.org`.

There are [potential reliability issues with the SKS keyservers](https://code.firstlook.media/the-death-of-sks-pgp-keyservers-and-how-first-look-media-is-handling-it), although in practice I'm not sure we're really affected by it.

The downsides to switching, as far as I can see:

1. `keys.openpgp.org` isn't federated, so not all of the keys we use can be found there; that would mean a simple one-off task to upload them.
2. the privacy feature of not sharing unverified email addresses is not supported by stock gnupg (including the version in homebrew). However it works fine with the version from gpg-suite, which is what we recommend anyway.

File: source/manual/create-a-gpg-key.html.md [hunk 1]
```diff
@@ -12,8 +12,6 @@ We use GPG keys to encrypt our secrets. Documentation for using your GPG key can
 
 Install `gpg` if you don't already have it. Use `brew install gpg-suite` to install the graphical [GPG Suite](https://gpgtools.org/).
 
-Once installed, you will likely have both `gpg` and `gpg2` on your machine. Always use `gpg2`.
```
Comment (ChrisBAshton) [inline source/manual/create-a-gpg-key.html.md:15]: Nice find 👍 I can confirm mine seems to be v2:

```
$ gpg --version
gpg (GnuPG/MacGPG2) 2.2.24
```

File: source/manual/create-a-gpg-key.html.md [hunk 2]
```diff
@@ -61,10 +61,10 @@ The key ID is `90E65803`, and the fingerprint is `37CC 021A C5C2
 
 ### Upload your GPG key to a keyserver
 
-Before doing this, make sure that your default keyserver is `http://keyserver.ubuntu.com/`. You can do this by changing the default keyserver in `~/.gnupg/gpg.conf`:
+Before doing this, make sure that your default keyserver is `https://keyserver.ubuntu.com/`. You can do this by changing the default keyserver in `~/.gnupg/gpg.conf`:
 
 ```
-keyserver http://keyserver.ubuntu.com/
+keyserver https://keyserver.ubuntu.com/
```
Comment (ChrisBAshton) [inline source/manual/create-a-gpg-key.html.md:65]: Good change 👍 as you've said elsewhere in the PR, this server is problematic, and we've had more luck recently with keys.openpgp.org. This is what I have in my `~/.gnupg/dirmngr.conf`, which works well for me:

```
keyserver hkps://keys.openpgp.org
```

I agree it is probably worth changing our recommended server to this keys.openpgp.org one, in a new commit that justifies the reasoning.

**Review by ChrisBAshton — COMMENTED** [review]
Looking good to me 👍 just a few thoughts/questions below.

File: source/manual/create-a-gpg-key.html.md [hunk 3]
```diff
@@ -50,14 +50,14 @@ gpg2 --fingerprint firstname.lastname@digital.cabinet-office.gov.uk
 Should look something like this.
 
 ```
-pub   2048R/90E65803 2013-02-08
-      Key fingerprint = 37CC 021A C5C2 4E27 C4D9  5735 9B0E 9DD1 90E6 5803
-      uid                  my name <my.name@digital.cabinet-office.gov.uk>
-      sub   2048R/FDD27DBE 2013-02-08
```
[the remark this answers is not in the document]
Comment (benjamineskola) [reply source/manual/create-a-gpg-key.html.md:56]: No, I only get one — I think any subkeys should be prefixed with `sub` (and actually, in the version I'm using it doesn't show any key ID for the subway at all).

I'm guessing the output format changed at least in part to discourage use of the 8-character version of the key ID, but it's a shame that it no longer seems to prefix the fingerprint with `fingerprint` like it used to.

File: source/manual/create-a-gpg-key.html.md [hunk 4]
```diff
@@ -59,10 +59,12 @@ and the key ID is `90E65803` — the last 8 characters of the fingerprint.
 
 ### Upload your GPG key to a keyserver
 
-Before doing this, make sure that your default keyserver is `hkps://keys.openpgp.org/`. You can do this by changing the default keyserver in `~/.gnupg/gpg.conf`:
+Before doing this, make sure that your default keyserver is `hkps://keys.openpgp.org/`. You can do this by changing the default keyserver in `~/.gnupg/dirmngr.conf`:
 
 ```
 keyserver hkps://keys.openpgp.org/
+
+You should also ensure that there is no `keyserver` directive in `~/.gnupg/gpg.conf`.
```
Comment (ChrisBAshton) [inline source/manual/create-a-gpg-key.html.md:67]: This should be moved below the ``` on line 68.

**Review by ChrisBAshton — APPROVED** [review]
LGTM 👍  thanks!

**Pull request merged**